### ISSUE TITLE
Only synonym labels returned by Phenotype Profile Analysis autocomplete

### ISSUE THREAD

**Issue by mbrush** [monarch-initiative/monarch-legacy#1510]
Term synonym labels seem to be returned by autocomplete/suggest feature of the Phenotype Profile Analysis service, instead of the primary term label.  This causes confusion for users - esp if the synonym is  not intuitively the same as the primary label. Example task and workflow below.
 
 
**Task:** Do a phenotype profile analysis for the mouse [Ebf3 gene](https://monarchinitiative.org/gene/MGI:894289#phenotypes) with three phenotypes, using the  [Phenotype Profile Analysis service](https://monarchinitiative.org/analyze/phenotypes).
 
**Workflow:** 
1. Search for the names of each phenotype reported on the Ebf3 page, e.g. "abnormal sensory neuron innervation pattern"
2. The auto complete/suggest feature does not return a term with an exact  label match, but does return the term "abnormal sensory neuron projections".
3. Hover over this autocomplete suggestion shows that it has the same IRI as the term I was searching for, and further exploration tells me that  "abnormal sensory neuron projections" is a synonym for "abnormal sensory neuron innervation pattern" (see http://www.informatics.jax.org/vocab/mp_ontology/MP:0000968)
  
It is confusing that the label for this term displayed on the Monarch page for the mouse Ebf3 gene is the primary term label, and the term appearing in autocomplete is a synonym (and the primary label is not returned, even when type din exactly).  Note that I had this issue for all three of the phenotypes linked to the mouse Ebf3 gene - so the example above is not an isolated case.

**Comment by kshefchek** [comment]
Looks like the solr highlighter returns multiple matched fields within a document even with an exact match on the primary label, and we pick [the first in the list](https://github.com/monarch-initiative/monarch-app/blob/35163bcc46c6a454bddb2289cb954e8dac767066/lib/monarch/api.js#L2855).  For example when searching for "Depressed nasal ridge"

     "highlighting":{
    "HP:0000457":{
      "synonym_eng":["<em class=\"hilite\">Depressed</em> <em class=\"hilite\">nasal</em> dorsum",
        "Retruded <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>",
        "Recessed <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>",
        "<em class=\"hilite\">Depressed</em> dorsum of nose",
        "Retruded <em class=\"hilite\">nasal</em> dorsum",
        "Flat <em class=\"hilite\">nasal</em> dorsum",
        "Recessed <em class=\"hilite\">nasal</em> dorsum"],
      "synonym_std":["<em class=\"hilite\">Depressed</em> <em class=\"hilite\">nasal</em> dorsum",
        "Retruded <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>",
        "Recessed <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>",
        "<em class=\"hilite\">Depressed</em> dorsum of nose",
        "Retruded <em class=\"hilite\">nasal</em> dorsum",
        "Flat <em class=\"hilite\">nasal</em> dorsum",
        "Recessed <em class=\"hilite\">nasal</em> dorsum"],
      "definition_std":["Lack of prominence of the nose resulting from a posteriorly-placed <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>."],
      "label_std":["<em class=\"hilite\">Depressed</em> <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>"],
      "definition_eng":["Lack of prominence of the nose resulting from a posteriorly-placed <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>."],
      "label_eng":["<em class=\"hilite\">Depressed</em> <em class=\"hilite\">nasal</em> <em class=\"hilite\">ridge</em>"],
      "label_kw":["<em class=\"hilite\">Depressed nasal ridge</em>"]},

I'll take a look at various solr settings to see if this can be less arbitrary, maybe someone has experience with this? @kltm @DoctorBud

**Comment by kshefchek** [comment]
Maybe it's a hack but we could also do this in the js code by prioritizing any field with a single hilite.

**Comment by kltm** [comment]
Hacks are fine. You may also want to consider "longest" and the like. I'd have to double check the default we went for...
In the future, I think that it might be nice to just merge them though--something that we have been planning on for a while.
Essentially, you have a character string and and int string that is zeroed, every time a character has a hit, it increments; afterwards, you walk it and apply spans (possibly darkening on higher values).

**Comment by kshefchek** [comment]
The code here addresses this: https://github.com/biolink/ontobio/pull/152, specifically https://github.com/biolink/ontobio/pull/152/files#diff-0c54f4a4fbd4e61af1fdb288882d2735R382
